packages/react-pdf/src/renderer.js

```diff
@@ -89,24 +92,28 @@ const PDFRenderer = ReactFiberReconciler({
     text,
     rootContainerInstance,
     hostContext,
-    internalInstanceHandle,
+    internalInstanceHandle
   ) {
-    return {
-      text,
-      tag: 'TEXT',
-    };
+    // return text;
+    return createElement('TEXT', { content: 'TEXT' });
+    // return {
+    //   text,
+    //   tag: 'TEXT',
+    // };
   },
 
   commitTextUpdate(textInstance, oldText, newText) {
     textInstance.text = newText;
   },
 
   appendChild(parentInstance, child) {
-    const index = parentInstance.children.indexOf(child);
-    if (index !== -1) {
-      parentInstance.children.splice(index, 1);
-    }
-    parentInstance.children.push(child);
+    child.inject(parentInstance);
```

Comment:
Dit this the other way around, because of how yoga-layout works... Internally it would switch from child to parent, will check if I can easily make it work the other way around.